powerauth-java-cmd-lib/src/main/java/io/getlime/security/powerauth/lib/cmd/util/HttpUtil.java

```diff
@@ -29,15 +29,10 @@
  */
 public class HttpUtil {
 
-    /**
-     * Take HTTP headers and convert list in value to string by taking the first value from the list.
-     * @param headers Headers to be flattened.
-     * @return Map(String,String) of flattened HTTP headers.
-     */
-    public static Map<String, String> flattenHttpHeaders(Headers headers) {
+    public static Map<String, String> flattenHttpHeaders(HttpHeaders headers) {
         Map<String, String> result = new HashMap<>();
-        for (Header h : headers.all()) {
-            result.put(h.getName(), h.getValue());
+        for (Map.Entry<String, List<String>> entry : headers.entrySet()) {
+            result.put(entry.getKey(), entry.getValue().get(0));
```

Comment:
Use `getFirst` method ([link](https://docs.spring.io/spring/docs/current/javadoc-api/org/springframework/http/HttpHeaders.html#getFirst-java.lang.String-)), just to avoid the static analysis about impossible NPE/AIOOB warning later.